[Enhancement]Load Model Once and Reuse Across Multiple Tests in BWC Test

BWC tests are unstable with frequent random failures during model auto-loading in cluster upgrades, especially in rolling upgrade BWC tests. The issue arises because each test attempts to load an ML model into memory, increasing the likelihood of auto-loading failures as multiple models need to be redeployed. In reality, we only need three models: sparse embedding, text embedding, and text-image embedding. By loading only these three models and reusing them across tests, we can significantly reduce test failures caused by model auto-redeployment during cluster upgrades. This does not compromise the purpose of the BWC test since we are not validating how well the cluster upgrade handles multiple models in memory but rather focusing on testing the functionality of the neural plugin during cluster upgrade.


```
REPRODUCE WITH: gradlew ':qa:rolling-upgrade:testAgainstTwoThirdsUpgradedCluster' --tests "org.opensearch.neuralsearch.bwc.SemanticSearchIT.testSemanticSearch_E2EFlow" -Dtests.seed=3D4E586DC98C0331 -Dtests.security.manager=false -Dtests.bwc.version=2.12.0 -Dtests.locale=ja-JP -Dtests.timezone=Australia/South -Druntime.java=17
org.opensearch.neuralsearch.bwc.SemanticSearchIT > testSemanticSearch_E2EFlow FAILED
    org.opensearch.client.ResponseException: method [GET], host [http://[::1]:61211], URI [/_cluster/health?wait_for_nodes=3&timeout=60s&wait_for_status=green&cluster_manager_timeout=60s], status line [HTTP/1.1 408 Request Timeout]


Suite: Test class org.opensearch.neuralsearch.bwc.SemanticSearchIT
  2> REPRODUCE WITH: gradlew ':qa:rolling-upgrade:testAgainstTwoThirdsUpgradedCluster' --tests "org.opensearch.neuralsearch.bwc.SemanticSearchIT.testSemanticSearch_E2EFlow" -Dtests.seed=3D4E586DC98C0331 -Dtests.security.manager=false -Dtests.bwc.version=2.12.0 -Dtests.locale=ja-JP -Dtests.timezone=Australia/South -Druntime.java=17
  2> org.opensearch.client.ResponseException: method [GET], host [http://[::1]:61211], URI [/_cluster/health?wait_for_nodes=3&timeout=60s&wait_for_status=green&cluster_manager_timeout=60s], status line [HTTP/1.1 408 Request Timeout]
    {"cluster_name":"neuralSearchBwcCluster-rolling","status":"yellow","timed_out":true,"number_of_nodes":3,"number_of_data_nodes":3,"discovered_master":true,"discovered_cluster_manager":true,"active_primary_shards":19,"active_shards":38,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1,"delayed_unassigned_shards":0,"number_of_pending_tasks":0,"number_of_in_flight_fetch":0,"task_max_waiting_in_queue_millis":0,"active_shards_percent_as_number":97.43589743589743}
        at __randomizedtesting.SeedInfo.seed([3D4E586DC98C0331:8B5CF7F56E9F9CF6]:0)
    {"cluster_name":"neuralSearchBwcCluster-rolling","status":"yellow","timed_out":true,"number_of_nodes":3,"number_of_data_nodes":3,"discovered_master":true,"discovered_cluster_manager":true,"active_primary_shards":19,"active_shards":38,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1,"delayed_unassigned_shards":0,"number_of_pending_tasks":0,"number_of_in_flight_fetch":0,"task_max_waiting_in_queue_millis":0,"active_shards_percent_as_number":97.43589743589743}
        at __randomizedtesting.SeedInfo.seed([3D4E586DC98C0331:8B5CF7F56E9F9CF6]:0)
        at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:479)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:371)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:346)
        at app//org.opensearch.neuralsearch.BaseNeuralSearchIT.waitForClusterHealthGreen(BaseNeuralSearchIT.java:1286)
        at app//org.opensearch.neuralsearch.bwc.SemanticSearchIT.testSemanticSearch_E2EFlow(SemanticSearchIT.java:28)
  1> [2024-12-19T07:00:37,819][INFO ][o.o.n.b.SemanticSearchIT ] [testSemanticSearch_E2EFlow] before test
  1> [2024-12-19T07:00:37,819][INFO ][o.o.n.b.SemanticSearchIT ] [testSemanticSearch_E2EFlow] initializing REST clients against [http://127.0.0.1:60984, http://[::1]:60985, http://127.0.0.1:61210, http://[::1]:61211, http://[::1]:60760, http://127.0.0.1:60761]
  1> [2024-12-19T07:01:38,211][INFO ][o.o.n.b.SemanticSearchIT ] [testSemanticSearch_E2EFlow] after test
        at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:479)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:371)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:346)
        at app//org.opensearch.neuralsearch.BaseNeuralSearchIT.waitForClusterHealthGreen(BaseNeuralSearchIT.java:1286)
        at app//org.opensearch.neuralsearch.bwc.SemanticSearchIT.testSemanticSearch_E2EFlow(SemanticSearchIT.java:28)
  2> NOTE: leaving temporary files on disk at: D:\a\neural-search\neural-search\qa\rolling-upgrade\build\testrun\testAgainstTwoThirdsUpgradedCluster\temp\org.opensearch.neuralsearch.bwc.SemanticSearchIT_3D4E586DC98C0331-001
  2> NOTE: test params are: codec=Asserting(Lucene912): {}, docValues:{}, maxPointsInLeafNode=1042, maxMBSortInHeap=5.5[425](https://github.com/opensearch-project/neural-search/actions/runs/12398942653/job/34618623713?pr=1028#step:4:426)43137940153, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=ja-JP, timezone=Australia/South
  2> NOTE: Windows Server 2022 10.0 amd64/Azul Systems, Inc. 17.0.13 (64-bit)/cpus=4,threads=2,free=269216192,total=536870912
  2> NOTE: All tests run in this JVM: [BatchIngestionIT, HybridSearchIT, HybridSearchWithRescoreIT, KnnRadialSearchIT, MultiModalSearchIT, NeuralQueryEnricherProcessorIT, NeuralSparseSearchIT, NeuralSparseTwoPhaseProcessorIT, SemanticSearchIT]
```
### Case 1
```
ERROR][o.o.m.a.MLModelAutoReDeployer] [neuralSearchBwcCluster-rolling-1] Exception occurred when auto redeploying the model, model id is: UF3OSJQBUk6b2XcQnjx2, exception is: no eligible node found, skipping current model auto redeploy and starting next model redeploy!
»  java.lang.IllegalArgumentException: no eligible node found
»  	at org.opensearch.ml.action.deploy.TransportDeployModelAction.deployModel(TransportDeployModelAction.java:248) [opensearch-ml-2.14.0.0.jar:2.14.0.0]
»  	at org.opensearch.ml.action.deploy.TransportDeployModelAction.lambda$doExecute$4(TransportDeployModelAction.java:152) [opensearch-ml-2.14.0.0.jar:2.14.0.0]
»  	at org.opensearch.core.action.ActionListener$1.onResponse(ActionListener.java:82) [opensearch-core-2.14.0.jar:2.14.0]
»  	at org.opensearch.ml.model.MLModelManager.lambda$getModel$69(MLModelManager.java:1572) [opensearch-ml-2.14.0.0.jar:2.14.0.0]
»  	at org.opensearch.core.action.ActionListener$1.onResponse(ActionListener.java:82) [opensearch-core-2.14.0.jar:2.14.0]
»  	at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:115) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:109) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleResponse(TransportSingleShardAction.java:298) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleResponse(TransportSingleShardAction.java:284) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1505) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.NativeMessageHandler.doHandleResponse(NativeMessageHandler.java:420) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.NativeMessageHandler.handleResponse(NativeMessageHandler.java:412) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.NativeMessageHandler.handleMessage(NativeMessageHandler.java:167) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.NativeMessageHandler.messageReceived(NativeMessageHandler.java:119) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.InboundHandler.messageReceivedFromPipeline(InboundHandler.java:108) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.InboundHandler.inboundMessage(InboundHandler.java:100) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.TcpTransport.inboundMessage(TcpTransport.java:784) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.nativeprotocol.NativeInboundBytesHandler.forwardFragments(NativeInboundBytesHandler.java:157) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.nativeprotocol.NativeInboundBytesHandler.doHandleBytes(NativeInboundBytesHandler.java:94) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:143) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:119) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:95) [transport-netty4-client-2.14.0.jar:2.14.0]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:280) [netty-handler-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) [netty-common-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.109.Final.jar:4.1.109.Final]
»  	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
»   ↑ repeated 2 times ↑
» ERROR][o.o.m.a.MLModelAutoReDeployer] [neuralSearchBwcCluster-rolling-1] Exception occurred when auto redeploying the model, model id is: hkrOSJQBLic8uyMcsizY, exception is: no eligible node found, skipping current model auto redeploy and starting next model redeploy!
»  java.lang.IllegalArgumentException: no eligible node found
»  	at org.opensearch.ml.action.deploy.TransportDeployModelAction.deployModel(TransportDeployModelAction.java:248) [opensearch-ml-2.14.0.0.jar:2.14.0.0]
»  	at org.opensearch.ml.action.deploy.TransportDeployModelAction.lambda$doExecute$4(TransportDeployModelAction.java:152) [opensearch-ml-2.14.0.0.jar:2.14.0.0]
»  	at org.opensearch.core.action.ActionListener$1.onResponse(ActionListener.java:82) [opensearch-core-2.14.0.jar:2.14.0]
»  	at org.opensearch.ml.model.MLModelManager.lambda$getModel$69(MLModelManager.java:1572) [opensearch-ml-2.14.0.0.jar:2.14.0.0]
»  	at org.opensearch.core.action.ActionListener$1.onResponse(ActionListener.java:82) [opensearch-core-2.14.0.jar:2.14.0]
»  	at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:115) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:109) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleResponse(TransportSingleShardAction.java:298) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleResponse(TransportSingleShardAction.java:284) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1505) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.NativeMessageHandler.doHandleResponse(NativeMessageHandler.java:420) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.NativeMessageHandler.handleResponse(NativeMessageHandler.java:412) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.NativeMessageHandler.handleMessage(NativeMessageHandler.java:167) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.NativeMessageHandler.messageReceived(NativeMessageHandler.java:119) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.InboundHandler.messageReceivedFromPipeline(InboundHandler.java:108) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.InboundHandler.inboundMessage(InboundHandler.java:100) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.TcpTransport.inboundMessage(TcpTransport.java:784) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.nativeprotocol.NativeInboundBytesHandler.forwardFragments(NativeInboundBytesHandler.java:157) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.nativeprotocol.NativeInboundBytesHandler.doHandleBytes(NativeInboundBytesHandler.java:94) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:143) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:119) [opensearch-2.14.0.jar:2.14.0]
»  	at org.opensearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:95) [transport-netty4-client-2.14.0.jar:2.14.0]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:280) [netty-handler-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:[919](https://github.com/opensearch-project/neural-search/actions/runs/12682096283/job/35347320319?pr=1081#step:4:920)) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) [netty-transport-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) [netty-common-4.1.109.Final.jar:4.1.109.Final]
»  	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.109.Final.jar:4.1.109.Final]
»  	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
»   ↑ repeated 2 times ↑
```

### Case 2
```
Caused by: java.lang.NullPointerException: Cannot invoke "org.opensearch.ml.task.MLTaskCache.getMlTask()" because "taskCache" is null
»  	at org.opensearch.ml.task.MLTaskManager.remove(MLTaskManager.java:141) ~[?:?]
»  	at org.opensearch.ml.task.MLTaskManager.updateMLTask(MLTaskManager.java:296) ~[?:?]
»  	at org.opensearch.ml.task.MLTaskManager.updateMLTask(MLTaskManager.java:276) ~[?:?]
»  	at org.opensearch.ml.action.forward.TransportForwardAction.doExecute(TransportForwardAction.java:150) ~[?:?]
»  	at org.opensearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:220) ~[opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.action.support.TransportAction.execute(TransportAction.java:190) ~[opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:133) ~[opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:129) ~[opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.wlm.WorkloadManagementTransportInterceptor$RequestHandler.messageReceived(WorkloadManagementTransportInterceptor.java:63) ~[opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:108) ~[opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.NativeMessageHandler.handleRequest(NativeMessageHandler.java:278) [opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.NativeMessageHandler.handleMessage(NativeMessageHandler.java:146) [opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.NativeMessageHandler.messageReceived(NativeMessageHandler.java:126) [opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.InboundHandler.messageReceivedFromPipeline(InboundHandler.java:120) [opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.InboundHandler.inboundMessage(InboundHandler.java:112) [opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.TcpTransport.inboundMessage(TcpTransport.java:796) [opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.InboundBytesHandler.forwardFragments(InboundBytesHandler.java:137) [opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.InboundBytesHandler.doHandleBytes(InboundBytesHandler.java:77) [opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:124) [opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:113) [opensearch-2.17.0.jar:2.17.0]
»  	at org.opensearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:95) [transport-netty4-client-2.17.0.jar:2.17.0]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:280) [netty-handler-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) ~[netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1407) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:918) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) [netty-transport-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:994) [netty-common-4.1.112.Final.jar:4.1.112.Final]
»  	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.112.Final.jar:4.1.112.Final]
»  	at java.lang.Thread.run(Thread.java:1583) [?:?]
» ERROR][o.o.m.m.MLModelManager   ] [neuralSearchBwcCluster-rolling-0] No controller is deployed because the model 6oMrSZQB_veoRncfWHHV is expected not having an enabled model controller. Please use the create model controller api to create one if this is unexpected.
```

--- a/qa/rolling-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/BatchIngestionIT.java
+++ b/qa/rolling-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/BatchIngestionIT.java
@@ -21,7 +21,7 @@ public class BatchIngestionIT extends AbstractRollingUpgradeTestCase {
     private static final String EMBEDDING_FIELD_NAME = "passage_embedding";
 
     public void testBatchIngestion_SparseEncodingProcessor_E2EFlow() throws Exception {
-        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER, 90);
         String indexName = getIndexNameForTest();
         String sparseModelId = null;
         switch (getClusterType()) {

--- a/qa/rolling-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/MultiModalSearchIT.java
+++ b/qa/rolling-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/MultiModalSearchIT.java
@@ -30,7 +30,7 @@ public class MultiModalSearchIT extends AbstractRollingUpgradeTestCase {
     // Create Text Image Embedding Processor, Ingestion Pipeline and add document
     // Validate process , pipeline and document count in rolling-upgrade scenario
     public void testTextImageEmbeddingProcessor_E2EFlow() throws Exception {
-        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER, 90);
         switch (getClusterType()) {
             case OLD:
                 modelId = uploadTextImageEmbeddingModel();

--- a/qa/rolling-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/NeuralSparseSearchIT.java
+++ b/qa/rolling-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/NeuralSparseSearchIT.java
@@ -38,7 +38,7 @@ public class NeuralSparseSearchIT extends AbstractRollingUpgradeTestCase {
     // Create Sparse Encoding Processor, Ingestion Pipeline and add document
     // Validate process , pipeline and document count in rolling-upgrade scenario
     public void testSparseEncodingProcessor_E2EFlow() throws Exception {
-        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER, 90);
         switch (getClusterType()) {
             case OLD:
                 modelId = uploadSparseEncodingModel();

--- a/qa/rolling-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/SemanticSearchIT.java
+++ b/qa/rolling-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/SemanticSearchIT.java
@@ -25,7 +25,7 @@ public class SemanticSearchIT extends AbstractRollingUpgradeTestCase {
     // Create Text Embedding Processor, Ingestion Pipeline and add document
     // Validate process , pipeline and document count in rolling-upgrade scenario
     public void testSemanticSearch_E2EFlow() throws Exception {
-        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER, 90);
         switch (getClusterType()) {
             case OLD:
                 modelId = uploadTextEmbeddingModel();

--- a/src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java
+++ b/src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java
@@ -1285,13 +1285,18 @@ public abstract class BaseNeuralSearchIT extends OpenSearchSecureRestTestCase {
     }
 
     // Method that waits till the health of nodes in the cluster goes green
-    protected void waitForClusterHealthGreen(final String numOfNodes) throws IOException {
+    protected void waitForClusterHealthGreen(final String numOfNodes, final int timeoutInSeconds) throws IOException {
         Request waitForGreen = new Request("GET", "/_cluster/health");
         waitForGreen.addParameter("wait_for_nodes", numOfNodes);
         waitForGreen.addParameter("wait_for_status", "green");
-        waitForGreen.addParameter("cluster_manager_timeout", "60s");
-        waitForGreen.addParameter("timeout", "60s");
+        waitForGreen.addParameter("cluster_manager_timeout", String.format(LOCALE, "%ds", timeoutInSeconds));
+        waitForGreen.addParameter("timeout", String.format(LOCALE, "%ds", timeoutInSeconds));
         client().performRequest(waitForGreen);
+    }
+
+    // Method that waits till the health of nodes in the cluster goes green with default timeout value of 60
+    protected void waitForClusterHealthGreen(final String numOfNodes) throws IOException {
+        waitForClusterHealthGreen(numOfNodes, 60);
     }
 
     /**

